### PR TITLE
include perfect scroll support for message content.

### PR DESCRIPTION
**Pull request by rox163**
# minor
# Changelog
## Non breaking changes
- added perfect scroll in the root addon for dialog content.
- updated header+footer styles for scrolling
- using ember-prop-types

### CODE REVIEW

File: addon/components/frost-modal-dialog.js [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
-  layout: layout,
+export default Component.extend(PropTypeMixin, {
   tagName: 'frost-modal-dialog',
   classNames: ['frost-modal-dialog'],
 
+  propTypes: {
+    title: PropTypes.string,
+    type: PropTypes.string.isRequired,
+    confirmAlias: PropTypes.string,
+    onConfirmHandler: PropTypes.func
```
Comment (sandersky) [inline addon/components/frost-modal-dialog.js]: Not a big deal but I tend to keep these sorted alphabetically.

File: blueprints/ember-frost-modal-dialog/index.js [hunk 2]
```diff
@@ -12,7 +12,10 @@ module.exports = {
     return this.addAddonsToProject({
       packages: [
         {name: 'ember-frost-core', target: '>=0.8.0 <2.0.0'},
-        {name: 'ember-block-slots', target: '0.12.1'}
+        {name: 'ember-block-slots', target: '^0.12.2'},
+        {name: 'ember-perfectscroll', target: '^0.1.2'},
+        {name: 'ember-remodal', target: '^1.2.0'},
+        {name: 'ember-wormhole', target: '0.3.5'}
```
Comment (sandersky) [inline blueprints/ember-frost-modal-dialog/index.js:19]: You should add `ember-prop-types` to this list as well.

**Comment by sandersky** [comment]
:+1:

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-modal-dialog/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-modal-dialog)

File: addon/components/frost-modal-dialog.js [hunk 3]
```diff
@@ -1,14 +1,32 @@
+/* global Ps */
 import Ember from 'ember'
-import layout from '../templates/components/frost-modal-dialog'
 import computed from 'ember-computed-decorators'
+import PropTypeMixin, {PropTypes} from 'ember-prop-types'
```
Comment (juwara0) [inline addon/components/frost-modal-dialog.js:4]: I updated ember-prop-types to add itself to the component in an initializer. If you want you can change this to just import the PropTypes:

`import {PropTypes} from 'ember-prop-types'`

File: addon/components/frost-modal-dialog.js [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
 
 const {Component, assert} = Ember
 
-export default Component.extend({
-  layout: layout,
+export default Component.extend(PropTypeMixin, {
```
Comment (juwara0) [inline addon/components/frost-modal-dialog.js:8]: Based on the note above about not needing to add PropTypeMixin, if you want, you can change this declaration to:

`export default Component.extend({`

Comment (sandersky) [inline addon/components/frost-modal-dialog.js:8]: @juwara0 I believe within an addon you still need to explicitly use `PropTypeMixin` as I was running into issues when trying to not explicitly use it in an addon.

Comment (juwara0) [inline addon/components/frost-modal-dialog.js:8]: @sandersky Ok. Thanks.

**Comment by juwara0** [comment]
👍 

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-modal-dialog/pull-request/5/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-modal-dialog)

File: addon/components/frost-modal-dialog.js (hunk 4 above)
Comment (sglanzer-fire) [inline addon/components/frost-modal-dialog.js:8]: @sandersky Can you elaborate on the issues?  I just added it to `ember-frost-core` using the implicit syntax

**Pull request merged**